src/components/modal/Modal.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import classes from './Modal.module.scss';
+import classNames from 'classnames';
+
+export default class Modal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: true
+    };
+  }
+
+  handleClose = () =>  {
+    this.setState({
+      isOpen: false
+    });
+  }
+
+  handleKeyDown = (event) => {
+    if (event.keyCode === 27) {
+      this.handleClose();
+    }
+  }
+
+  handleClickAway = (event) =>  {
+    if (!this.modalDialog.contains(event.target)) {
+      this.handleClose();
+    }
+  }
+
+  setScrollLock = (isLocked) => {
+    document.body.style.overflow = isLocked ? "hidden" : 'auto';
+  }
+
+  cleanup = () => {
+    this.setScrollLock(false);
+
+    if (this.props.onClose) {
+      this.props.onClose();
+    }
+  }
+
+  componentDidMount() {
+    this.setScrollLock(true);
+
+    this.modalContainer.focus();
+  }
+
+  componentWillUnmount() {
+    this.cleanup();
+  }
+
+  render() {
+    const { isOpen } = this.state;
```

Comment:
Is this needed? Or can the modal be destroyed (unmount) when it's closed?

Reply:
Doing it manually is an anti-pattern in React (afaik). This is the proper way to do it and replicates a parent controlling the rendering of a child through a prop (`handleClose` can be called through the `ModalService`). React deals with the unmounting of it once the component is no longer needing to be rendered.